### PR TITLE
OP-626_close-button-in-patient-examination

### PR DESCRIPTION
**Pull request by mwithi**
See OP-626.

### CODE REVIEW

File: src/main/java/org/isf/examination/gui/PatientExaminationEdit.java [hunk 1]
```diff
@@ -1190,10 +1189,15 @@ private JSlider getJSliderHeight() {
 		if (jSliderHeight == null) {
 			jSliderHeight = new JSlider(ExaminationParameters.HEIGHT_MIN, ExaminationParameters.HEIGHT_MAX, ExaminationParameters.HEIGHT_INIT);
 			jSliderHeight.addChangeListener(e -> {
-				int value = jSliderHeight.getValue();
-				jTextFieldHeight.setText(String.valueOf(value));
-				patex.setPex_height(value);
-				updateBMI();
+				JSlider source = (JSlider)e.getSource();
+		        if (!source.getValueIsAdjusting()) {
+		        	System.out.println("change!");
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/examination/gui/PatientExaminationEdit.java:1194]: Not sure you want to leave this in production.

File: src/main/java/org/isf/examination/gui/PatientExaminationEdit.java [hunk 2]
```diff
@@ -1204,11 +1208,15 @@ private ScaledJSlider getJSliderWeight() {
...
 			jSliderWeight = new ScaledJSlider(ExaminationParameters.WEIGHT_MIN, ExaminationParameters.WEIGHT_MAX, ExaminationParameters.WEIGHT_STEP, ExaminationParameters.WEIGHT_INIT);
 			jSliderWeight.addChangeListener(e -> {
-				double value = jSliderWeight.getScaledValue();
-				//double weight = (double) value / 10;
-				jTextFieldWeight.setText(String.valueOf(value));
-				patex.setPex_weight(value);
-				updateBMI();
+				JSlider source = (JSlider)e.getSource();
+		        if (!source.getValueIsAdjusting()) {
+		        	System.out.println("change!");
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/examination/gui/PatientExaminationEdit.java:1213]: And here too

File: src/main/java/org/isf/examination/gui/PatientExaminationEdit.java [hunk 3]
```diff
@@ -1190,10 +1189,15 @@ private JSlider getJSliderHeight() {
...
 			jSliderHeight.addChangeListener(e -> {
-				int value = jSliderHeight.getValue();
-				jTextFieldHeight.setText(String.valueOf(value));
-				patex.setPex_height(value);
-				updateBMI();
+				JSlider source = (JSlider)e.getSource();
+		        if (!source.getValueIsAdjusting()) {
+		        	System.out.println("change!");
+		        	int value = jSliderHeight.getValue();
+					jTextFieldHeight.setText(String.valueOf(value));
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/examination/gui/PatientExaminationEdit.java:1196]: formatting?

File: src/main/java/org/isf/examination/gui/PatientExaminationEdit.java [hunk 4]
```diff
@@ -1204,11 +1208,15 @@ private ScaledJSlider getJSliderWeight() {
...
-				double value = jSliderWeight.getScaledValue();
-				//double weight = (double) value / 10;
-				jTextFieldWeight.setText(String.valueOf(value));
-				patex.setPex_weight(value);
-				updateBMI();
+				JSlider source = (JSlider)e.getSource();
+		        if (!source.getValueIsAdjusting()) {
+		        	System.out.println("change!");
+		        	double value = jSliderWeight.getScaledValue();
+					jTextFieldWeight.setText(String.valueOf(value));
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/examination/gui/PatientExaminationEdit.java:1215]: formatting?

File: src/main/java/org/isf/examination/gui/PatientExaminationEdit.java (hunk 1 above)
Comment (mwithi) [reply src/main/java/org/isf/examination/gui/PatientExaminationEdit.java:1194]: Sure not! thanks!

File: src/main/java/org/isf/examination/gui/PatientExaminationEdit.java (hunk 3 above)
Comment (mwithi) [reply src/main/java/org/isf/examination/gui/PatientExaminationEdit.java:1195]: Thanks!

File: src/main/java/org/isf/examination/gui/PatientExaminationEdit.java (hunk 4 above)
Comment (mwithi) [reply src/main/java/org/isf/examination/gui/PatientExaminationEdit.java:1213]: Thanks!

**Comment by dbmalkovsky** [comment]
I believe the "modification indicator" needs additional logic (and this is why I had suggested we don't do it at this time).   For example, if I click one of the check boxes, say next to "Heart Rate", and then click the same checkbox again to uncheck it without making any other modifications, when I click on "Close" it asks me whether I want to save the changes.   There aren't really any differences.

File: src/main/java/org/isf/examination/gui/PatientExaminationEdit.java [hunk 5]
```diff
@@ -65,6 +65,7 @@
 import javax.swing.OverlayLayout;
 import javax.swing.ScrollPaneConstants;
 import javax.swing.SwingConstants;
+import javax.swing.Timer;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/examination/gui/PatientExaminationEdit.java:68]: I don't believe this import is used.

File: src/main/java/org/isf/examination/gui/PatientExaminationEdit.java [hunk 6]
```diff
@@ -1330,11 +1336,24 @@ private JButton getJButtonDelete() {
 		return jButtonDelete;
 	}
 	
-	private JButton getJButtonCancel() {
+	private JButton getJButtonClose() {
 		if (jButtonCancel == null) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/examination/gui/PatientExaminationEdit.java:1340]: shouldn't the button be renamed to `jButtonClose`?

File: src/main/java/org/isf/examination/gui/PatientExaminationEdit.java (hunk 5 above)
Comment (mwithi) [reply src/main/java/org/isf/examination/gui/PatientExaminationEdit.java:68]: Thanks!

File: src/main/java/org/isf/examination/gui/PatientExaminationEdit.java (hunk 6 above)
Comment (mwithi) [reply src/main/java/org/isf/examination/gui/PatientExaminationEdit.java:1340]: Yep, thank you!

**Comment by mwithi** [comment]
> I believe the "modification indicator" needs additional logic (and this is why I had suggested we don't do it at this time). For example, if I click one of the check boxes, say next to "Heart Rate", and then click the same checkbox again to uncheck it without making any other modifications, when I click on "Close" it asks me whether I want to save the changes. There aren't really any differences.

I know about this, I've tried to implement more rigorous logics but the way fields are updated doesn't make things simple. Anyway I think the current logic it's a good compromise for a reasonable usage: how many users will just check and uncheck one parameter only (as the only change) and expecting not to be warned about changes? In contrary, to have at least a check is what a common user would expect. We have to consider real cases... majority will just open the window, set parameters and save... do you agree?

**Comment by mwithi** [comment]
> Yes that is the typical usage.

approved?

**Comment by dbmalkovsky** [comment]
LGTM given the limitation.

**Pull request merged**